### PR TITLE
Avoid two leading slashes in the request to create a new file

### PR DESCRIPTION
**Pull request by aboktor**
## Description
When using PFT+POP auth with ODSP, ODSP fails to validate the `p` field in the token when we have two leading slashes in the path. This is likely a bug in ODSP but it does result in cryptic 401 failure of the request.

## Breaking Changes
None

## Reviewer Guidance

None

### CODE REVIEW

**Review by AndreiZe — APPROVED** [review]
LGTM

**Comment by tylerbutler** [comment]
@aboktor please make sure you get review and approval from someone on the framework team before merge.

File: packages/drivers/odsp-driver/src/createFile.ts
Comment (vladsud) [inline packages/drivers/odsp-driver/src/createFile.ts:167]: make it path: string | undefined and do the check or undefined here once, instead of each time this function is called.

**Review by vladsud — APPROVED** [review]
Worth double-checking if such check should be done a bit earlier in the stack, i.e. is it the only place that is affected?

**Pull request merged**
